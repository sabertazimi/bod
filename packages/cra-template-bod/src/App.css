.app-header {
  @apply flex flex-col items-center justify-center;

  min-height: 100vh;
  font-size: calc(10px + 2vmin);
}

.app-link {
  color: rgb(112 76 182);
}

.app-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .app-logo {
    animation: app-logo-spin infinite 20s linear;
  }
}

@keyframes app-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
