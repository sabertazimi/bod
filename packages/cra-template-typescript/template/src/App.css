.app-header {
  @apply flex flex-col items-center justify-center;

  min-height: 100vh;
  font-size: calc(10px + 2vmin);
  color: white;
  background-color: #282c34;
}

.app-link {
  color: #61dafb;
}

.app-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .app-logo {
    animation: app-logo-spin infinite 20s linear;
  }
}

@keyframes app-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
