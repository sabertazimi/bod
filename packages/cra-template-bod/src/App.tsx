import { Route, Routes } from 'react-router-dom';
import './App.css';
import { Counter } from './containers';
import Logo from './logo.svg';

const App = (): JSX.Element => {
  return (
    <div className="bg-white text-center">
      <header className="app-header">
        <Logo className="app-logo" title="logo" />
        <Routes>
          <Route path="/" element={<Counter />} />
        </Routes>
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <span>
          <span>Learn </span>
          <a
            className="app-link"
            href="https://react.dev"
            target="_blank"
            rel="noopener noreferrer"
          >
            React
          </a>
          <span>, </span>
          <a
            className="app-link"
            href="https://redux.js.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Redux
          </a>
          <span>, </span>
          <a
            className="app-link"
            href="https://sabertazimi.github.io/bod"
            target="_blank"
            rel="noopener noreferrer"
          >
            Bod App
          </a>
        </span>
      </header>
    </div>
  );
};

export default App;
