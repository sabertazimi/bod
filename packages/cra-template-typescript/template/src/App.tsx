import Logo from './logo.svg'
import './App.css'

/**
 * Renders the main application component.
 * @returns {JSX.Element} The rendered App component.
 */
function App() {
  return (
    <div className="text-center">
      <header className="app-header">
        <Logo className="app-logo" title="logo" />
        <p>
          Edit
          {' '}
          <code>src/App.tsx</code>
          {' '}
          and save to reload.
        </p>
        <a
          className="app-link"
          href="https://react.dev"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  )
}

export default App
