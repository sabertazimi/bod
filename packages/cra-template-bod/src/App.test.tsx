import { render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import App from './App';
import store from './stores';

describe('App', () => {
  test('should render correctly (snapshot)', () => {
    const tree = create(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render accessibility guidelines (AXE)', async () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );

    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });

  test('renders learn React, Redux and Bod links', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );

    expect(screen.getByText(/learn/i)).toBeInTheDocument();
    expect(screen.getByText(/react/i)).toBeInTheDocument();
    expect(screen.getByText(/redux/i)).toBeInTheDocument();
    expect(screen.getByText(/bod/i)).toBeInTheDocument();
  });
});
