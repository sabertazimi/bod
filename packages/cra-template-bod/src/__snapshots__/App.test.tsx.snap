// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly (snapshot) 1`] = `
<div
  className="bg-white text-center"
>
  <header
    className="app-header"
  >
    <svg
      className="app-logo"
      title="logo"
    />
    <div>
      <div
        className="row"
      >
        <button
          aria-label="Decrement value"
          className="button"
          onClick={[Function]}
        >
          -
        </button>
        <span
          className="value"
        >
          0
        </span>
        <button
          aria-label="Increment value"
          className="button"
          onClick={[Function]}
        >
          +
        </button>
      </div>
      <div
        className="row"
      >
        <input
          aria-label="Set increment amount"
          className="textbox"
          onChange={[Function]}
          value="2"
        />
        <button
          className="button"
          onClick={[Function]}
        >
          Add Amount
        </button>
        <button
          className="asyncButton"
          onClick={[Function]}
        >
          Add Async
        </button>
        <button
          className="button"
          onClick={[Function]}
        >
          Add If Odd
        </button>
      </div>
    </div>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <span>
      <span>
        Learn 
      </span>
      <a
        className="app-link"
        href="https://react.dev"
        rel="noopener noreferrer"
        target="_blank"
      >
        React
      </a>
      <span>
        , 
      </span>
      <a
        className="app-link"
        href="https://redux.js.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Redux
      </a>
      <span>
        , 
      </span>
      <a
        className="app-link"
        href="https://sabertazimi.github.io/bod"
        rel="noopener noreferrer"
        target="_blank"
      >
        Bod App
      </a>
    </span>
  </header>
</div>
`;
